src/handler.ts

```diff
@@ -59,6 +59,7 @@ abstract class AbstractFileHandler implements IMessageFilter {
    */
   constructor(manager: IContentsManager) {
     this._manager = manager;
+    document.addEventListener('focus', this._onFocus.bind(this), true);
```

Comment:
consider making `_onFocus` a member function:

```typescript
private _onFocus = (evt: Event) => {
  // do stuff with `this`
};
```

This way it can be removed as a listener later on if needed.